reconcile/external_resources/manager.py

```diff
@@ -0,0 +1,423 @@
+
+        if current_job:
+            logging.info("Removing previous Job instance")
+            if not dry_run:
+                self.oc.delete(self.namespace, "Job", job_name)
+
+        logging.info("Creating Job to reconcile external resource")
+        if not dry_run:
+            job_spec = self._build_job_spec(
+                key=key, image=image, input=input, action=action
+            )
+            job = OpenshiftResource(
+                job_spec, "QONTRACT_INTEGRATION", "QONTRACT_INTEGRATION_VERSION"
+            )
+            dj = job.annotate()
+            self.oc.apply(self.namespace, dj)
+
+
+class ExternalResourcesManager:
+    def __init__(
+        self,
+        oc: OCCli,
+        cluster: str,
+        namespace: str,
+        state_manager: ExternalResourcesStateManager,
+        settings: ExternalResourcesSettings,
+        modules: Mapping[str, ExternalResourceModule],
+        impl: ExternalResourcesImpl,
+        dry_run: bool = False,
+    ) -> None:
+        self.state_mgr = state_manager
+        self.oc = oc
+        self.cluster = cluster
+        self.namespace = namespace
+        self.settings = settings
+        self.modules = modules
+        self.impl = impl
+        self.dry_run = dry_run
+        self.providers = ObjectFactory[ExternalResourceFactory]()
+        self.providers.register_factory("aws", AWSExternalResourceFactory(settings))
+
+    def get_external_resources(
+        self, namespaces: list[NamespaceV1]
+    ) -> list[ExternalResourceSpec]:
+        resources: list[ExternalResourceSpec] = []
+
+        # PoC Filter
+        for ns in namespaces:
+            if ns.name != "external-resources-poc":
+                continue
+            if ns.external_resources:
+                for erp in ns.external_resources:
+                    if (
+                        type(erp) is NamespaceTerraformProviderResourceAWSV1
+                        and erp.resources
+                    ):
+                        for er in erp.resources:
+                            if type(er) is NamespaceTerraformResourceRoleV1:
+                                # Using this class is far from ideal as it uses
+                                # generic dicts. This might need improved logic.
+                                ers = ExternalResourceSpec(
+                                    provision_provider=erp.provider,
+                                    provisioner=erp.provisioner.dict(),
+                                    resource=er.dict(),
+                                    namespace=ns.dict(),
+                                )
+                                resources.append(ers)
+
+                break
+
+        return resources
+
+    def resource_needs_reconciliation(
+        self,
+        resource: ExternalResource,
+        state: ExternalResourceState,
+    ) -> bool:
+        if state.status == ResourceStatus.NOT_EXISTS:
```

Comment:
you can use [match ... case](https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching) here.

Reply:
Done, This changed recently but I managed to use match-case.